### PR TITLE
Point out that the payload size could decrease

### PR DESCRIPTION
**Pull request by ianswett**
And this could make retransmitting frames complex.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -3374,9 +3374,11 @@ priorities specified by the application indicate otherwise (see
 
 Even though a sender is encouraged to assemble frames containing up-to-date
 information every time it sends a packet, it is not forbidden to retransmit
-copies of frames from lost packets.  A receiver MUST accept packets containing
-an outdated frame, such as a MAX_DATA frame carrying a smaller maximum data than
-one found in an older packet.
+copies of frames from lost packets.  A sender which retransmits copies of frames
+must deal with decreases in available payload size due to change in packet
+number length, connection ID length, and Path MTU.  A receiver MUST accept
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md]:
```suggestion
number length, connection ID length, and path MTU.  A receiver MUST accept
```

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -3374,9 +3374,11 @@ priorities specified by the application indicate otherwise (see
 
 Even though a sender is encouraged to assemble frames containing up-to-date
 information every time it sends a packet, it is not forbidden to retransmit
-copies of frames from lost packets.  A receiver MUST accept packets containing
-an outdated frame, such as a MAX_DATA frame carrying a smaller maximum data than
-one found in an older packet.
+copies of frames from lost packets.  A sender which retransmits copies of frames
+must deal with decreases in available payload size due to change in packet
```
Comment (erickinnear) [inline draft-ietf-quic-transport.md]: Lowercase `must` makes me read this oddly, how about: 
```suggestion
needs to handle decreases in available payload size due to change in packet
```

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -3374,9 +3374,11 @@ priorities specified by the application indicate otherwise (see
 
 Even though a sender is encouraged to assemble frames containing up-to-date
 information every time it sends a packet, it is not forbidden to retransmit
-copies of frames from lost packets.  A receiver MUST accept packets containing
-an outdated frame, such as a MAX_DATA frame carrying a smaller maximum data than
-one found in an older packet.
+copies of frames from lost packets.  A sender which retransmits copies of frames
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md]:
```suggestion
copies of frames from lost packets.  A sender that retransmits copies of frames
```

CMOS

File: draft-ietf-quic-transport.md [hunk 4]
```diff
@@ -3374,9 +3374,11 @@ priorities specified by the application indicate otherwise (see
...
 Even though a sender is encouraged to assemble frames containing up-to-date
 information every time it sends a packet, it is not forbidden to retransmit
-copies of frames from lost packets.  A receiver MUST accept packets containing
-an outdated frame, such as a MAX_DATA frame carrying a smaller maximum data than
-one found in an older packet.
+copies of frames from lost packets.  A sender that retransmits copies of frames
+needs to handle decreases in available payload size due to change in packet
+number length, connection ID length, and path MTU.  A receiver MUST accept
+packets containing an outdated frame, such as a MAX_DATA frame carrying a
+smaller maximum data than one found in an older packet.
```
Comment (mikkelfj) [inline draft-ietf-quic-transport.md]: I don't understand this text: I can see old data arriving in new packets and you can't generally which is the latest valid, except when you can, such as MAX_DATA. You don't want to accept old MAX_DATA frames (or at least not change behaviour based on it).

**Pull request merged**